### DISCUSSION TITLE
How to change the position of "spawn"?

### DISCUSSION THREAD

**Discussion by Cebolitoss** [google/model-viewer#4044] [Q&A]
I would like to place the object a bit down of the surface, and not exactly in the surface. As half of it is "buried". I would move it in the z axis (up-down) just a bit.

I see the orientation and the scale attributes but I don't see the move/location attribute. How can I do that?

Other thing is, my 3D models are being placed not really in the middle of the screen/scene. It places the object always a bit on top of the screen. And it came up a bit for from the user. How can I settle it to always load in a short distance of the user's location?

**Comment by Cebolitoss** [comment]
In the video you can see the placement that I'm talking about.
It should be placed in the carpet close to me, and not there close to the fridge, far way from me. 

And in the image bellow you can also see the origin of my 3d model. It is on [0,0] coordinates.

https://user-images.githubusercontent.com/117696201/211741828-666adbe6-485c-4c6e-89fd-54c4e26e1691.mp4

![image](https://user-images.githubusercontent.com/117696201/211742244-8b8547a0-dd20-4d86-9843-2fa6244e92e7.png)

**Reply by Cebolitoss** [reply]
I tried placing the object under the "surface" on my software and transforming it, according to the origin. So the origin, that before, was bellow the whole object, now, it is above the whole object.


Even if I do that, it spawn in the "ground" again on AR.

![image](https://user-images.githubusercontent.com/117696201/211745151-d04fa5d9-3841-40c6-8431-510e855519a7.png)

**Reply by elalish** [reply]
We don't currently have an API for this; the lowest part of the object's bounding box is always placed on the floor. However, this has been asked for before and it's a good idea: https://github.com/google/model-viewer/discussions/3985#discussioncomment-4411984

Interested in designing an API or making a PR? Also, if you want to place something below the floor, how are you hoping it will render? We don't currently use AR occlusions, though it could probably be enabled with some work. Still, I'm curious if the accuracy is adequate for your use case?

**Reply by Cebolitoss** [reply]
So I would like to try to create some kind of feature that shows for example the pipe in a shower.
So the shower would be fixed in the wall, and I could show in some way the point of the pipe that it is attached to the shower, but this part of the shower, would be inside the wall. So I could make the pipe a bit transparent or someth like that, so the user would know that what is transparent in his shower, this part would be inside the wall, and the normal part, without transparency, this would be the part of the shower that is visible, outside the wall.

Other use case for me would be for example a tap, and the pipes, under the sink. I mean not the whole pipe, but the piece that comes together with the tap, or the shower.

**Reply by Cebolitoss** [reply]
Also I didn't get that, when you asked "Interested in designing an API or making a PR?"
What would be a PR? I mean, what PR stands for?

I'm trying to put together some features in an AR experience using model-viewer and/or other tools that allow me to do this!
And I have lots of questions related to the model-viewer and coding it!

I might be starting other discussions trying to reach my goal. Is it ok?
Questions related to shadows in the parts of the object that are hidden/transparent (alpha factor = 1), that still appear in the ground.
Starting and stopping animations, animations running in loops and animations that might run once and stop at some part of it. For example a button that allows the user to open a lid when he clicks on it, and to close the lid when he clicks on it again.
Showing and hiding the measures in the AR experience. But these are all questions to new discussions.

**Reply by elalish** [reply]
I like your idea, and conveniently it doesn't require any AR occlusion. PR stands for "pull request" which is the standard Git parlance for contributing code to a project like model-viewer. I'm not going to have to implement this feature in the near term, so if you want it, adding it yourself is the best way (if you're comfortable with forking, building, and modifying our TS codebase). You can start the PR with just the documentation of your feature and I'll review it and let you know if the API is a good shape to start implementing .

Yes, transparent objects cast opaque shadows. It's a known limitation and will take quite a bit of work to change, so it's not a priority. The animation stuff you describe can all be accomplished with our current API; look through the docs and examples . Feel free to ask more questions, but please read the docs and existing discussions first to see if you can find an answer.
